### ISSUE TITLE
RA consumes all the RAM with recursive scan

### ISSUE THREAD

**Issue by sergiobenrocha2** [libretro/RetroArch#3966]
Ubuntu 14.04 desktop, RA is taking 4 GB of the RAM when I try to scan! (in a clean config). Edit: forgot to say, the test folder has +120 zipped roms. Edit2: the folder has a subfolder, and it has others folders... something like it:

Folder0
--- Folder1
------- rom A
------- rom B
------- zipped file C (with some files)
------- ...
------- Folder1.1
------------- rom
------------- file
------------- ...
------- Folder1.2
------------- rom
------------- ...
--- rom 1
--- rom 2
--- ...

**Comment by sergiobenrocha2** [comment]
Are there any way to disable recursive scan?

**Comment by i30817** [comment]
Eh, my scan (of a 350 gb compressed mount squash file) took 3 hours, but it never ran out of memory to me. Although i did skip the 90 gb subdirectory holding dos games.

It might be the difference was that the compression was external and it wasn't retroarch that had to do the partial uncompression to check the md5sums.

**Comment by sergiobenrocha2** [comment]
```
.
├── Advanced Busterhawk Gleylancer (Japan).zip
├── Aladdin (USA).zip
├── Alien Soldier (Japan).zip
├── Alisia Dragoon (USA).zip
├── Ayrton Senna's Super Monaco GP II (USA) (En,Ja).zip
├── Battletoads-Double Dragon (USA).zip
├── Beyond Oasis (USA).zip
├── Bio Hazard Battle (USA, Europe).zip
├── Castle of Illusion Starring Mickey Mouse (USA, Europe).zip
├── Castlevania - Bloodlines (USA).zip
├── Comix Zone (USA).zip
├── Contra - Hard Corps (USA, Korea).zip
├── Desert Strike - Return to the Gulf (USA, Europe).zip
├── Dr. Robotnik's Mean Bean Machine (USA).zip
├── Dune II - The Battle for Arrakis (Europe).zip
├── Dynamite Headdy (USA, Europe).zip
├── Earthworm Jim (USA).zip
├── Ecco the Dolphin (USA, Europe).zip
├── Ecco - The Tides of Time (USA).zip
├── ESWAT - City Under Siege (USA, Europe).zip
├── Eternal Champions (USA).zip
├── Flashback (Europe) (Rev A).zip
├── Flink (Europe).zip
├── Gaiares (Japan, USA).zip
├── Gain Ground (World) (Rev A).zip
├── General Chaos (USA, Europe).zip
├── Golden Axe II (World).zip
├── Golden Axe (World) (v1.1).zip
├── Gunstar Heroes (USA).zip
├── Hacks & Prototypes
│   ├── (4401)UMKT_HACK_18_FIXED.zip
│   ├── (4401)UMKT_HACK_18+TRACKS_FIXED.zip
│   ├── accuracy tests
│   │   ├── md_test
│   │   │   ├── 512PAL.BIN
│   │   │   ├── 960PAL.bin
│   │   │   ├── MD1536.bin
│   │   │   ├── mdtrans.bin
│   │   │   ├── memtest_68k.bin
│   │   │   ├── PAL512.bin
│   │   │   ├── SHLTEST2.BIN
│   │   │   ├── SHLTEST.BIN
│   │   │   ├── STETEST2.bin
│   │   │   ├── stetest.bin
│   │   │   ├── vctest.bin
│   │   │   ├── VDPTEST.bin
│   │   │   ├── wbug.bin
│   │   │   ├── wtest.bin
│   │   │   ├── wtest_i1.bin
│   │   │   └── wtest_i2.bin
│   │   ├── md_test.zip
│   │   ├── SpriteMaskingTestRom.bin
│   │   ├── SpriteMaskingTestRom.bin.gz
│   │   ├── VDPFIFOTesting
│   │   │   ├── asm68k.exe
│   │   │   ├── build.cmd
│   │   │   ├── cram.bin
│   │   │   ├── TestBlankingFlags.asm
│   │   │   ├── TestCodeAndAddressRegisters.asm
│   │   │   ├── TestCPWritePendingReset.asm
│   │   │   ├── TestCRAMByteswapping.asm
│   │   │   ├── TestDMABusyFlagDMACopy.asm
│   │   │   ├── TestDMABusyFlagDMADisabledCopy.asm
│   │   │   ├── TestDMABusyFlagDMADisabledFill.asm
│   │   │   ├── TestDMABusyFlagDMAEnableToggleCopy.asm
│   │   │   ├── TestDMABusyFlagDMAEnableToggleFill.asm
│   │   │   ├── TestDMABusyFlagDMAFill.asm
│   │   │   ├── TestDMABusyFlagDMATransfer.asm
│   │   │   ├── TestDMACopy.asm
│   │   │   ├── TestDMACopy - Backup.asm
│   │   │   ├── TestDMACopyLengthRegUpdate.asm
│   │   │   ├── TestDMACopySourceRegUpdate.asm
│   │   │   ├── TestDMACopyToCRAMV2.asm
│   │   │   ├── TestDMACopyToVSRAMV2.asm
│   │   │   ├── TestDMAFillControlPortWrites.asm
│   │   │   ├── TestDMAFillDataPortWritesCRAM.asm
│   │   │   ├── TestDMAFillDataPortWritesVRAM.asm
│   │   │   ├── TestDMAFillDataPortWritesVSRAM.asm
│   │   │   ├── TestDMAFillFIFOUsage.asm
│   │   │   ├── TestDMAFillLengthRegUpdate.asm
│   │   │   ├── TestDMAFillSourceRegUpdate.asm
│   │   │   ├── TestDMAFillToCRAM.asm
│   │   │   ├── TestDMAFillToCRAMV2.asm
│   │   │   ├── TestDMAFillToVRAM.asm
│   │   │   ├── TestDMAFillToVRAMV2.asm
│   │   │   ├── TestDMAFillToVSRAM.asm
│   │   │   ├── TestDMAFillToVSRAMV2.asm
│   │   │   ├── TestDMATransferBusLock.asm
│   │   │   ├── TestDMATransferLengthRegUpdate.asm
│   │   │   ├── TestDMATransferSourceAddressWrapping.asm
│   │   │   ├── TestDMATransferSourceRegUpdate.asm
│   │   │   ├── TestDMATransferToCRAM.asm
│   │   │   ├── TestDMATransferToCRAMWrapping.asm
│   │   │   ├── TestDMATransferToVRAM.asm
│   │   │   ├── TestDMATransferToVRAMWrapping.asm
│   │   │   ├── TestDMATransferToVSRAM.asm
│   │   │   ├── TestDMATransferToVSRAMWrapping.asm
│   │   │   ├── TestDMATransferUsingFIFO.asm
│   │   │   ├── TestFIFOBufferSize.asm
│   │   │   ├── TestFIFOFullDMATransfer.asm
│   │   │   ├── TestFIFOWaitStates.asm
│   │   │   ├── TestFIFOWaitStates - Old testing.asm
│   │   │   ├── TestFIFOWriteToInvalidTarget.asm
│   │   │   ├── TestHVCounterLatch.asm
│   │   │   ├── TestMode4RegisterMask.asm
│   │   │   ├── TestPartialCPWrites.asm
│   │   │   ├── TestReadTarget01100.asm
│   │   │   ├── TestReadTargetSwitching.asm
│   │   │   ├── TestReadWriteOrder.asm
│   │   │   ├── TestRegisterWriteBit13Masked.asm
│   │   │   ├── TestRegisterWritesAffectCodeRegister.asm
│   │   │   ├── TestSeparateFIFOReadWriteBuffer.asm
│   │   │   ├── TestVRAMByteswapping.asm
│   │   │   ├── TestVSRAMByteswapping.asm
│   │   │   ├── TestVSRAMDataCache.asm
│   │   │   ├── TestVSRAMDataCache - Copy.asm
│   │   │   ├── TestVSRAMDataCacheNew.asm
│   │   │   ├── TestVSRAMDataCache_old.asm
│   │   │   ├── VDPFIFOTesting.asm
│   │   │   ├── VDPFIFOTesting.bin
│   │   │   ├── vram.bin
│   │   │   └── VRAMTestDataBlock.bin
│   │   └── VDPFIFOTesting.zip
│   ├── Amy_In_Sonic_1_Rev_1.8.zip
│   ├── AngryBridsdemo.bin.zip
│   ├── BadApple.bin
│   ├── BadApple_p1.bin
│   ├── BadApple_p2.bin
│   ├── BASIC KERNEL.zip
│   ├── Cave-Story-MD.bin
│   ├── Donkey Kong 1999.zip
│   ├── Flux Mega CD.zip
│   ├── FM Editor for GENWINDOWS V0[1].1.zip
│   ├── How to Use Map.zip
│   ├── Joystick Test Program.zip
│   ├── Knuckles_the_Echidna_in_Sonic_the_Hedgehog.zip
│   ├── Miracle Piano Teaching System.zip
│   ├── Multitap - IO Sample Program.zip
│   ├── Pier Solar (JUE) (Beta).bin.zip
│   ├── Portal_Gun_in_Sonic_2_(snkenjoi)_(SHC2015).zip
│   ├── Rockman X3.zip
│   ├── S2heroes-v0.06h.7z
│   ├── Sonic_2_and_Knuckles_Long_Version_v.1.8.zip
│   ├── Sonic_2_Delta_2_V0.10a.zip
│   ├── Sonic_2_Delta_V0.23.zip
│   ├── Sonic_2_Long_Version_v.1.8.1.zip
│   ├── Sonic 2 Retro Remix_sage_2011.zip
│   ├── Sonic_2_-_The_Hybridization_Project.zip
│   ├── Sonic3C_130810.zip
│   ├── Sonic & Knuckles (S2K chip) (Prototype 0618 - Jun 18, 1994, 9.07) (hidden-palace.org).bin
│   ├── Sonic_&_Knuckles_(S2K_chip)_(Prototype_0618_-_Jun_18,_1994,_9.07)_(hidden-palace.org).zip
│   ├── Sonic Megamix v3.0.zip
│   ├── Sonic the Hedgehog 2 (beta).zip
│   ├── Sonic_The_Hedgehog_2_(Simon_Wai_Prototype)_-!-.zip
│   ├── Sonic_Thrash_-_Second_Public_Release.zip
│   ├── STH_-_WI.zip
│   ├── Street Fighter 2 Turbo (Beta).zip
│   ├── Tails_in_Sonic_1_v2.1.1.zip
│   ├── titan-overdrivemegademo-v1.1-final.zip
│   └── YM 2612 Editor.zip
├── Hellfire (USA).zip
├── Herzog Zwei (USA, Europe).zip
├── Jungle Strike (USA, Europe).zip
├── Kid Chameleon (USA, Europe).zip
├── Landstalker (USA).zip
├── Lion King, The (World).zip
├── Lost Vikings, The (USA).zip
├── Mega Man - The Wily Wars (Europe).zip
├── Mega Turrican (USA).zip
├── Michael Jackson's Moonwalker (World) (Rev A).zip
├── Monster World IV (Japan).zip
├── Musha Aleste - Full Metal Fighter Ellinor (Japan).zip
├── NBA Jam - Tournament Edition (World).zip
├── NHL '94 (USA, Europe).zip
├── Phantasy Star II (Brazil).zip
├── Phantasy Star III - Generations of Doom (Brazil).zip
├── Phantasy Star IV (USA).zip
├── Pier Solar and the Great Architects (World) (En,Es,Pt) (Rev C) (Unl).iso -> ../Sega CD/Pier Solar and the Great Architects (World) (Unl)/Pier Solar and the Great Architects (World) (Unl).iso
├── Pier Solar and the Great Architects (World) (En,Es,Pt) (Rev C) (Unl).md
├── Pirates! Gold (USA).zip
├── Populous (USA).zip
├── Pulseman (Japan).zip
├── QuackShot Starring Donald Duck ~ QuackShot - I Love Donald Duck - Guruzia Ou no Hihou (World) (v1.1).zip
├── Ranger X (USA).zip
├── Red Zone (USA, Europe).zip
├── Revenge of Shinobi, The (USA, Europe) (Rev B).zip
├── Rings of Power (USA, Europe).zip
├── Ristar (USA, Europe) (September 1994).zip
├── Road Rash II (USA, Europe) (v1.2).zip
├── Rocket Knight Adventures (USA).zip
├── Rolo to the Rescue (USA, Europe).zip
├── Shadow Dancer - The Secret of Shinobi (World).zip
├── Shadowrun (USA).zip
├── Shining Force II (USA).zip
├── Shining Force (USA).zip
├── Shinobi III - Return of the Ninja Master (USA).zip
├── Simpsons, The - Bart vs. the Space Mutants (USA, Europe) (Rev A).zip
├── Sonic 3D Blast ~ Sonic 3D Flickies' Island (USA, Europe).zip
├── Sonic & Knuckles + Sonic The Hedgehog 3 (World).zip
├── Sonic & Knuckles (World).zip
├── Sonic The Hedgehog 2 (World) (Rev A).zip
├── Sonic The Hedgehog (Japan, Korea).zip
├── Sonic The Hedgehog Spinball (USA).zip
├── Splatterhouse 3 (USA).zip
├── Streets of Rage 2 (USA).zip
├── Streets of Rage 3 (USA).zip
├── Super Hang-On (World) (En,Ja) (Rev A).zip
├── Super Street Fighter II (USA).zip
├── Target Earth (USA).zip
├── Teenage Mutant Ninja Turtles - The Hyperstone Heist (USA).zip
├── Thunder Force III (Japan, USA).zip
├── Thunder Force IV (Japan).zip
├── Tiny Toon Adventures - Acme All-Stars (USA, Korea).zip
├── Tiny Toon Adventures - Buster's Hidden Treasure (USA).zip
├── ToeJam & Earl (World) (Rev A).zip
├── Urban Strike (USA, Europe).zip
├── Vectorman (USA, Europe).zip
├── Wonder Boy in Monster World (USA, Europe).zip
├── World of Illusion Starring Mickey Mouse and Donald Duck (USA, Korea).zip
├── X-Men 2 - Clone Wars (USA, Europe).zip
├── YuYu Hakusho - Sunset Fighters (Brazil).zip
├── Zero Wing (Europe).zip
└── Zombies Ate My Neighbors (USA).zip
```

**Comment by sergiobenrocha2** [comment]
wow, it's getting almost 5 GB now

**Comment by sergiobenrocha2** [comment]
@bparker06 I found what's triggering it, it's a zip inside another zip, try this file: https://bitbucket.org/eke/genesis-plus-gx/downloads/md_test.zip

**Comment by bparker06** [comment]
Fixed in 7be7568. Thanks

**Comment by sergiobenrocha2** [comment]
wow, that was fast. Thanks!
